framework/src/problems/FEProblemBase.C

```diff
@@ -5061,6 +5061,10 @@ FEProblemBase::addMultiApp(const std::string & multi_app_name,
     parameters.set<SystemBase *>("_sys") = _aux.get();
   }
 
+  if (!_app.parameters().get<bool>("use_legacy_fixed_point_execute_on") &&
+      !parameters.isParamSetByUser("execute_on"))
+    parameters.set<ExecFlagEnum>("execute_on", true) = FixedPointSolve::EXEC_FIXEDPOINT_BEGIN;
```

Comment:
This is not the right way to change the default execution of MultiApps. 
If the user explicitly sets the execute_on to its default with this, the behavior of the solve changes

Reply:
This is used to achieve the goal that I mentioned earlier: with or without the line in MooseTestApp.C, all tests can pass. If in the future we remove this legacy, we can change the default value in `MultiApp::validParams` directly.

Reply:
Passing the tests is good but the documentation says TIMESTEP_BEGIN and the reality is FIXEDPOINT_BEGIN with this change. We cant have that

Reply:
Is improving the docstring like `when use_legacy_fixed_point_execute_on is false, the default execute_on for MultiApps changes to fixedpoint_begin` ok?